packages/block-editor/src/store/actions.js

```diff
@@ -218,6 +218,29 @@ export function toggleSelection( isSelectionEnabled = true ) {
 	};
 }
 
+function getBlocksWithAutoApplyStyles( blocks, blockEditorSettings ) {
+	return blocks.map( ( block ) => {
+		const autoApplyBlockStyles = get( blockEditorSettings, [ 'autoApplyBlockStyles' ], {} );
```

Comment:
`autoApplyBlockStyles` should be assigned one level up, in `getBlocksWithAutoApplyStyles`.